### ISSUE TITLE
Support streaming compaction in NG

### ISSUE THREAD

**Issue by domodwyer** [influxdata/influxdb_iox#4324]
Currently the compactor reads all record batches into memory during compaction - it would be nice to stream directly into parquet files on disk to reduce the memory usage of the compactor.

**Comment by NGA-TRAN** [comment]
I hope you will cover #4101

**Comment by alamb** [comment]
Additional thoughts: the success critera would be to avoid the unbounded stream that was introduced: https://github.com/influxdata/influxdb_iox/pull/4325

**Comment by domodwyer** [comment]
Yes removing the unbounded channels would be my intention as part of this ticket 👍

**Comment by alamb** [comment]
Another potential issue is that we buffer the entire plan output into memory prior to writing to parquet

https://github.com/influxdata/influxdb_iox/blob/f9e3495/compactor/src/compact.rs#L731-L734

**Comment by domodwyer** [comment]
Yes this is what I'll be addressing - I'll also undo the buffering I added in https://github.com/influxdata/influxdb_iox/pull/4325 to get us compacting.
